addons/payment/models/payment_acquirer.py

```diff
@@ -277,14 +277,34 @@ def _create_missing_journal_for_acquirers(self, company=None):
             journals += acquirer.journal_id
         return journals
 
+    @api.model
+    def _update_journal(self, vals):
+        '''
+        Update the payment method of the journal depending on the field 'payment_flow' and 'save_token'.
+        If these fields signify that the payment is electronic, add the electronic payment
+        method to the journal. Otherwise, remove the electronic payment method.
+        '''
+        method = self.env.ref('payment.account_payment_method_electronic_in')
+        if not method: return
+        for record in self:
+            if not record.journal_id: continue
+            if (vals.get('payment_flow', record.payment_flow) == 's2s' or
+                vals.get('save_token', record.save_token) in ('ask', 'always')):
+
+                record.journal_id.inbound_payment_method_ids |= method
+            else:
+                record.journal_id.inbound_payment_method_ids -= method
+
     @api.model
     def create(self, vals):
         image_resize_images(vals)
+        self._update_journal(vals)
```

Comment:
wait a minute... That's plain wrong!

you can't call `_update_journal` on `self` before creating the record and then use `self` within the method and expect to have good results there.

you need to do something like
```python
    def create(self, vals):
        image_resize_images(vals)
        res = super(PaymentAcquirer, self).create(vals)
        res._update_journal()
        return res
```